wemake_python_styleguide/visitors/ast/functions.py

```diff
@@ -189,7 +191,10 @@ def _check_floating_nan(self, node: ast.Call) -> None:
         if not isinstance(node.args[0], (ast.Str, ast.Bytes)):
             return
 
-        if node.args[0].s.lower() in frozenset(('nan', b'nan')):
+        if not isinstance(node.func, ast.Name) or node.func.id != 'float':
```

Comment:
We have https://github.com/wemake-services/wemake-python-styleguide/blob/master/wemake_python_styleguide/logic/tree/functions.py#L13 just for this case.